martin's suggestions. (#154)
xmartlabs/blog: _posts/2021-07-30-gong-introduction.md

---
layout: post
title: Kickstart your Android Project with Gong App Template
date: 2021-08-04 10:00:00

> tags: [Android App Template, Android Base project, Android Compose Base Project, Xmartlabs Android App, Jetpack Compose App Template, Jetpack Compose Base Project, Android Development, Kotlin Development]

Reviewer: I think we should have `Jetpack Compose` as a separate tag too

Author: @sante please add it.

twitter_image: /images/gong-introduction/bannerTwitter.png
permalink: /blog/kickstart-your-new-android-project-with-gong-app-template/
---

<p style="text-align:left;">
<img src="/images/gong-introduction/bannerInterno.png" alt="Project banner">
</p>

Starting a software project is not an easy task and Android is no exception, there's a bunch of complex decisions to make such as code conventions, libraries, continuous integrations, environments management, app architecture, and best practices just to mention a few.

Each time we kick off a project we're forced to write this boilerplate code and typically we end up using the same project structure, libraries, code style, helpers over and over again, which is good to be consistent in the form we build apps but it's also a naivete issue to spend time building the same base project.

In this blog post, we introduce Gong, the Android App Template we use to kickstart every Android app we build. Gong was fully developed and it's actively maintained by our Android team, and it's the result of years of experience in creating well-written and maintainable Android code using Kotlin and recently Jetpack Compose.

# What exactly is Gong?

> [Gong][gong] is our Android App Template to effortlessly kickstart Android projects and get us quickly ready to implement product functionalities. By using Gong, in a matter of seconds you can create the base project for your android app, the same base project we use at Xmartlabs, you only have to clone the repo and run a provided script to set up the project name, namespace, and few more data about your particular app.

Reviewer: It's not necessary to clone the repo, the script does that for you
```suggestion
[Gong][gong] is our Android App Template to effortlessly kickstart Android projects and get us quickly ready to implement product functionalities. By using Gong, in a matter of seconds you can create the base project for your android app, the same base project we use at Xmartlabs, you only have to run command to fetch and execute a provided script to set up the project name, package name, and optionally your github repository's URL to have it set up from the get-go.
```

Reviewer: Also, the set up part is mentioned again below

Apart from providing a top-notch architecture and several out of the box project development solutions and tools that speed up Android app development, Gong enforces the usage of the same code conventions, adopt the same project structure, follow platform best practices, and use pretty much the same third-party libraries throughout the whole Android team.

Gong was intended to help Android teams to adopt elegant, clean, and maintainable Android code, use state-of-the-art components and libraries, and quickly get into coding valuable app functionality with virtually no effort and in no time.

# Kickstarting your project using Gong

Creating your app base code using Gong is super simple, you just have to clone the Gong repo and run a script that prompts you to set up your project information such as project and package name! That's all!

If you are interested, there is a step-by-step guide on how to initialize the script and provide such information on Gong's Github readme page.

> Once you provide the project's information, Gong automatically renames and refactors the template code according to your naming conventions (you never need to do any manual refactoring). You can even provide your Git remote URL so the project is set up pointing to your remote repository.

Reviewer: I don't know about naming conventions. I would say it changes the name of things but not following the conventions of the user because we don't know them. The naming conventions are ours, it's just the names which contain `Gong` that get modified

Author: elimina el `according to your naming conventions`  @sante

Reviewer: It's already done

Let's see how easy it's below:

<p style="text-align:center;">
<img style="width: 100%" src="/images/gong-introduction/howto.gif" alt="How to start using it">
</p>

# The Generated base Android project

> Once the script execution finishes, you can open the generated Android project, which was named <YourProjectName>.xyz.

Reviewer: ```suggestion
Once the script execution finishes, you can open the generated Android project, which is ready for you to work on.
```

permalink: /blog/kickstart-your-new-android-project-with-gong-base-project/
---

> <p style="text-center;">

Author: ```suggestion
<p style="text-align:center;">
```

> <p style="text-align:left;">

Author: ```suggestion
<p style="text-center;">
```

# What exactly is Gong?

[Gong][gong] is our Android App Template to effortlessly kickstart Android projects and get us quickly ready to implement product features.
By using Gong, in a matter of seconds, you can create the base project for your android app, the same base project we use at Xmartlabs.

Gong is entirely written in [Kotlin][kotlin], and it leverages the language's capabilities and modern syntax (if you're still programming Android apps in [Java][Java], give Kotlin a try, you won't regret it 😉).
It also uses [Jetpack Compose][compose], the new declarative toolkit for creating UI in Android.

In short, Gong is a base project which gives you a good structure, well-constructed architecture, good practices, and modern features and toolkits with virtually no effort and in no time, so you can start implementing features right away.

# Kickstarting your project using Gong

Creating your app base code using Gong is super simple. You just have to run a command to fetch and execute a script that asks for your project information, such as project and package name. That's all!

> The command is available on Gong's Github readme page, along with instructions to do the setup manually if you prefer to do so.

Reviewer 2: Add a link pointing to that section https://github.com/xmartlabs/gong#setup

<img src="/images/gong-introduction/bannerInterno.png" alt="Project banner">
</p>

Starting a software project is not an easy task, and Android is no exception.

> There's a bunch of complex decisions to make, such as code conventions, libraries, continuous integrations, environments management, app architecture, and best practices, to mention a few.

Reviewer 2: maybe a different order? architectures, frameworks, libraries. 
```suggestion
There's a bunch of complex decisions to make, such as code conventions, app architecture, frameworks, libraries, continuous integrations, and best practices, to mention a few.
```
